Yoast/wpseo-woocommerce
# Wordpress SEO/11543 cannot read property replace

## Summary

This PR can be summarized in the following changelog entry:

* The short product description assessment in the SEO analysis is now correctly updated whenever the description gets changed inside the Text-based editor (previously only in the Visual editor).

## Relevant technical choices:

* Moved some existing logic to a new class for getting the text from the current editor and sending the short product description to the YoastSEO.js web worker for analysis when it changes.

## Test instructions

This PR can be tested by following these steps:

* Start your local WordPress development server and login.
* Install the latest WooCommerce plugin.
* Make sure that you have the Yoast SEO plugin installed.
* Make sure that you have this repo (`wpseo-woocommerce`) locally cloned and up and running, on this branch.
* Go to the _Products_ page of WooCommerce. (You can find it in the sidebar of the admin area).
* Click on  the button 'Add New' (next to the Products page title).
* Navigate to the _Short Product Description_ editor, located in a meta box near the bottom of the age.
* Open the browser's development console.
* Switch the _Short Product Description_ editor to Text mode.
* Reload the page.
* [ ] You should *not* see an error in the console.
* Note the text of the product description assessment in the SEO analysis.
 It should read: `The short description for this product is too short.`.
* Type in 50 words of text in the _Short Product Description_ Text editor.
* [ ] The assessment should read: `Your short description has a good length.`.
* Type in 100 words of text.
* [ ] The assessment should read: `The short description for this product is too long.`.
* Switch to the Visual editor.
* Remove half of the text.
* [ ] The assessment should read: `Your short description has a good length.`.
* Delete the text entirely.
* [ ] The assessment should read: `The short description for this product is too short.`.

Fixes https://github.com/Yoast/wordpress-seo/issues/11543
